addons/website/models/website_menu.py

```diff
@@ -166,6 +166,24 @@ def make_tree(node):
         menu = menu_id and self.browse(menu_id) or website.menu_id
         return make_tree(menu)
 
+    def contains_record_url(self):
+        """
+        Check if the menu or one of its children contains a record URL.
+
+        :return: True if the menu or one of its children contains a record URL.
+        """
+        # We want to limit the number of sql queries so we KISS with the
+        # following check.
+        if self.url and self.url.startswith('/') and self.url.count('/') > 1:
+            last_part = self.url.split('/')[-1]
+            dash_parts = last_part.split('-')
+            if len(dash_parts) > 1 and all(dash_parts):
+                return True
+        for child in self.child_id:
+            if child.contains_record_url():
+                return True
+        return False
+
```

Comment:
Still not sure this is the good solution @rdeodoo. Does fixing the issue deserve to impact negatively the perf ? I think I can do something to do all this in one single SQL query by using the ORM the optimal way but are you sure it worth it ? Adding one SQL query to render a page to fix this bug seems a bit overkill IMO. Maybe I should go back to the JS solution.